PHPCompatibility/Sniffs/Constants/TemporarilyRemovedConstantsSniff.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2023 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Sniffs\Constants;
+
+use PHPCompatibility\Helpers\MiscHelper;
+use PHPCompatibility\Helpers\ScannedCode;
+use PHPCompatibility\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHPCSUtils\Utils\MessageHelper;
+
+/**
+ * Detect use of PHP native global constants which have been removed and re-added.
+ *
+ * PHP version All
+ *
+ * @since 10.0.0
+ */
+class TemporarilyRemovedConstantsSniff extends Sniff
+{
+
+    /**
+     * A list of removed & re-added PHP Constants.
+     *
+     * Note: PHP Constants are case-sensitive!
+     *
+     * @since 10.0.0
+     *
+     * @var array(string => array(string => float))
+     */
+    protected $constants = [
+        'T_BAD_CHARACTER' => [
+            'removed' => 7.0,
+            'last_absent' => 7.3,
+            're-added' => 7.4,
+        ],
+    ];
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @since 10.0.0
+     *
+     * @return array
+     */
+    public function register()
+    {
+        return [
+            \T_STRING,
+        ];
+    }
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @since 10.0.0
+     *
+     * @param File $phpcsFile The file being scanned.
+     * @param int  $stackPtr  The position of the current token in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens       = $phpcsFile->getTokens();
+        $constantName = $tokens[$stackPtr]['content'];
+
+        if (isset($this->constants[$constantName]) === false) {
+            return;
+        }
+
+        if (MiscHelper::isUseOfGlobalConstant($phpcsFile, $stackPtr) === false) {
+            return;
+        }
+
+        $itemArray = $this->constants[$constantName];
+
+        if (
+            ScannedCode::shouldRunOnOrAbove($itemArray['removed']) &&
+            ScannedCode::shouldRunOnOrBelow($itemArray['last_absent'])
```

Comment:
Nitpick (and yes, I'll sort out that we'll enforce that in the future - https://github.com/PHPCSStandards/PHPCSDevCS/pull/36).
```suggestion
            ScannedCode::shouldRunOnOrAbove($itemArray['removed'])
            && ScannedCode::shouldRunOnOrBelow($itemArray['last_absent'])
```

Reply:
Great, when that's released I'll update and run `phpcbf`.
Edit: I'll manually adjust this for now.